opensquirrel/decomposer/general_decomposer.py

```diff
@@ -32,25 +55,37 @@ def check_gate_replacement(gate: Gate, replacement_gates: Iterable[Gate]) -> Non
         msg = f"replacement for gate {gate.name} does not preserve the quantum state"
         raise ValueError(msg)
 
+    if qc is not None:
```

Comment:
I guess `qc` should be `circuit`, but then the methods that are called on `circuit` are the methods of the `PhaseMap` class, i.e., not of the Circuit class.

I'm still confused what the code block (of this if-statement) actually does. I get the idea, but I feel that this might actually _introduce_ relative phases, instead of mitigating them...

Can you better explain in the Confluence page what is being implemented and how that prevents a global phase difference in the gates (between original and replacement), from resulting in a relative phase between the qubits, because that bit is still unclear to me. To put it another way: How can a global phase difference in a gate replacement, lead to a relative phase difference. As far as I see it, this can only happen if a replaced gate (that differs by a global phase from the original) is made into a controlled gate, but this doesn't happen anywhere in the compilation process (yet).